File: LC/C++/Easy/2016.cpp
class Solution {
public:
    int maximumDifference(vector<int>& nums) {
        int curr_low = INT_MAX;
        int ans = 0;

        for (int num : nums) {
            curr_low = min(curr_low, num);
            ans = max(ans, num - curr_low);
        }

        return ans != 0 ? ans : -1;
    }
};

File: LC/Python/Easy/2016.py
class Solution:
    def maximumDifference(self, nums: List[int]) -> int:
        curr_low = inf
        ans = 0
        
        for num in nums:
            curr_low = min(curr_low, num)
            ans = max(ans, num - curr_low)
      
        return ans if ans != 0 else -1